plugwise/smile.py

```diff
@@ -215,6 +215,13 @@ def _get_device_data(self, dev_id):
             if power_data is not None:
                 device_data.update(power_data)
 
+        # Elga doesn't use intended_cental_heating_state to show the generic heating state
+        if "heating" in device_data and not device_data["heating"]:
```

Comment:
Secondary condition could fail if there is no heating in `device_data`, so it's either redundant or error prone. Isn't this better to (re)write as

```
if "c_heating" in device_data: # assuming heating is there only when c_heating is discovered should be safe?
  if device_data["c_heating"] and not device_data["heating"]:
    # insert setting heating to true + popping
```

Reply:
Yes, that's also a good , and shorter, option :)
The `central_heating_state` key is always there, next to the `intended_central_heating_state` (being always there).